src/components/core/geoChart/geoChartDataLayers.ts

```diff
@@ -63,8 +64,14 @@ function createPushpinSizeOptions(
     return sizeOptions;
 }
 
-function createPushpinFilter(selectedSegmentItem: string): mapboxgl.Expression {
-    return ["==", selectedSegmentItem, getExpressionByBucketName(SEGMENT_BY)];
+export function createPushpinFilter(selectedSegmentItems: string[]): mapboxgl.Expression {
+    return [
+        "match",
+        getExpressionByBucketName(SEGMENT_BY),
+        selectedSegmentItems.length > 0 ? selectedSegmentItems : [EMPTY_SEGMENT_VALUE],
```

Comment:
It could be just `selectedSegmentItems.length ? selectedSegmentItems : [EMPTY_SEGMENT_VALUE],`